src/test/bridges/notional/NotionalE2E.t.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {BridgeTestBase} from "./../../aztec/base/BridgeTestBase.sol";
+import {NotionalBridgeContract} from "../../../bridges/notional/NotionalBridge.sol";
+import {IWrappedfCashFactory} from "../../../interfaces/notional/IWrappedfCashFactory.sol";
+import {NotionalViews} from "../../../interfaces/notional/INotionalViews.sol";
+import {AztecTypes} from "../../../aztec/libraries/AztecTypes.sol";
+
+contract NotionalE2ETest is BridgeTestBase {
+    address public constant ETH = 0x0000000000000000000000000000000000000000;
+    uint16 public constant ETH_CURRENCY_ID = 1;
+    // The reference to the stability pool bridge
+    NotionalBridgeContract private bridge;
+    address private stabilityPool;
+    IWrappedfCashFactory public constant FCASH_FACTORY =
+        IWrappedfCashFactory(0x5D051DeB5db151C2172dCdCCD42e6A2953E27261);
+    NotionalViews public constant NOTIONAL_VIEW = NotionalViews(0x1344A36A1B56144C3Bc62E7757377D288fDE0369);
+    // To store the id of the stability pool bridge after being added
+    uint256 private id;
+
+    function setUp() public {
+        // Deploy a new stability pool bridge and set approvals
+        bridge = new NotionalBridgeContract(address(ROLLUP_PROCESSOR));
+
+        // Use the label cheat-code to mark the address in the traces
+        vm.label(address(bridge), "NotionalBridge");
+
+        // Impersonate the multi-sig to add a new bridge
+        vm.startPrank(MULTI_SIG);
+
+        // List the staking-bridge with a gasLimit of 350k
+        ROLLUP_PROCESSOR.setSupportedBridge(address(bridge), 1000000);
+
+        vm.stopPrank();
+
+        // Fetch the id of the notional bridge
+        id = ROLLUP_PROCESSOR.getSupportedBridgesLength();
+    }
+
+    // @dev In order to avoid overflows we set _depositAmount to be uint96 instead of uint256.
+    function testFullDepositWithdrawalE2E(uint256 _depositAmount) public {
+        vm.assume(_depositAmount > uint256(1e16));
+        vm.assume(_depositAmount < uint256(1e21));
```

Comment:
Please replace all the `vm.assume(...)` calls you are using in the tests with bound(...) form forge-std. This change is good to do because `vm.assume(...)` is not ideal when the amount of excluded values are too high --> the fuzzer then executes too few tests. Another tip is that you can make `_depositAmount` be a smaller type than is `uint256` (e.g. `uint64`). This way you can tell the fuzzer to use smaller values as well.